doc/manual/methods.rst

```diff
@@ -440,6 +440,198 @@ can also constrain type parameters of methods::
 The ``same_type_numeric`` function behaves much like the ``same_type``
 function defined above, but is only defined for pairs of numbers.
 
+
+Design Patterns with Parametric Methods
+---------------------------------------
+
+While complex dispatch logic is not required for performance or usability,
+sometimes it can be the best way to express some algorithm.
+Here are a few common design patterns that come up sometimes when using dispatch in this way.
+
+1. Extracting the type parameter from a super-type
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here is the correct code template for returning the element-type ``T``
+of any arbitrary subtype of ``AbstractArray``::
+
+    abstract AbstractArray{T, N}
+    eltype(::Type{AbstractArray}) = Any
+    eltype{T}(::Type{AbstractArray{T}}) = T
+    eltype{T, N}(::Type{AbstractArray{T, N}}) = T
+    eltype{A<:AbstractArray}(::Type{A}) = eltype(supertype(A))
+
+One common foible here is to try to extract the type-parameter directly::
```

Comment:
"foible" isn't a very common word for non-native speakers. Maybe find something easier to understand (here and below)?